### PR TITLE
refactor html page comparison test cases

### PR DESCRIPTION
**Pull request by chengren311**
1. leverage assertj to compare two strings ignore white spaces
2. add a new method called ignoreCaseContains in ExecutionFlowViewTest.java for more readability.

### CODE REVIEW

File: azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java [hunk 1]
```diff
@@ -122,8 +121,7 @@ public void createSuccessEmail() throws Exception {
         this.executableFlow, this.message, this.azkabanName, this.scheme, this.clientHostname,
         this.clientPortNumber));
     assertEquals("Flow 'mail-creator-test' has succeeded on unit-tests", this.message.getSubject());
-    assertEquals(read("successEmail.html").replaceAll("\\s+", ""),
-        this.message.getBody().replaceAll("\\s+", ""));
+    assertThat(read("successEmail.html")).isEqualToIgnoringWhitespace(this.message.getBody());
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: Thanks. This looks much better.

Is http://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/AbstractCharSequenceAssert.html#isEqualToNormalizingWhitespace(java.lang.CharSequence) more appropriate?

Comment (chengren311) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: isEqualToNormalizingWhitespace requires a base normalized string as argument. but here we don't have such string.

Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: Got it. It's unfortunate. If you feel like, file an enhancement request for assertj?

Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: According to the implementation:

  private boolean areEqualNormalizingWhitespace(CharSequence actual, CharSequence expected) {
    if (actual == null) return expected == null;
    checkCharSequenceIsNotNull(expected);
    return normalizeWhitespace(actual).equals(normalizeWhitespace(expected));
  }

It seems like this method should work. Have you tried it?

Comment (chengren311) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: yes,     
assertThat("a   b").isEqualToNormalizingWhitespace("ab") fails

Comment (chengren311) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: sorry,  assertThat("a  b").isEqualToNormalizingWhitespace("a b") works, but i'm not sure assertThat(read("successEmail.html")).isEqualToNormalizingWhitespace(this.message.getBody()) fails

Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/executor/mail/DefaultMailCreatorTest.java]: I tried it. Here is the difference.

```
<h2 style="color:#FF0000"> Execution '-1' of flow 'mail-creator-test' of project 'test-project' has encountered a failure on unit-tests</h2>This flow is set to complete all currently running jobs before stopping. <table> <tr> <td>Start Time</td> <td>2016/07/17 11:54:11 EEST</td> </tr> <tr> <td>End Time</td> <td>N/A</td> </tr> <tr> <td>Duration</td> <td>10 sec</td> </tr> <tr> <td>Status</td> <td>FAILED_FINISHING</td> </tr> </table><a href="http://localhost:8081/executor?execid=-1">mail-creator-test Execution Link</a><h3> Reason</h3> <ul> <li><a href="http://localhost:8081/executor?execid=-1&job=test-job">Failed job 'test-job' Link</a> </li> </ul>
<h2 style="color:#FF0000"> Execution '-1' of flow 'mail-creator-test' of project 'test-project' has encountered a failure on unit-tests</h2>This flow is set to complete all currently running jobs before stopping.<table><tr><td>Start Time</td><td>2016/07/17 11:54:11 EEST</td></tr><tr><td>End Time</td><td>N/A</td></tr><tr><td>Duration</td><td>10 sec</td></tr><tr><td>Status</td><td>FAILED_FINISHING</td></tr></table><a href="http://localhost:8081/executor?execid=-1">mail-creator-test Execution Link</a><h3>Reason</h3><ul><li><a href="http://localhost:8081/executor?execid=-1&job=test-job">Failed job 'test-job' Link</a></li></ul>
```
The actual output doesn't separate some elements with white spaces. It's not ideal and makes the test less strict.
However, I am willing to defer an improvement for now given other priorities. 

Thanks

**Pull request merged**